source/plugin/offline-userplayer-data.rst

```diff
@@ -0,0 +1,79 @@
+=======================================
+Accessing data of offline Users/Players 
+=======================================
+
+In almost every plugin it's necessary to get access to player information even when the player is offline.
+Since ``Sponge.getServer().getPlayer()`` does only return Players who are online one needs to find another solution.
+
+Some plugins store the relevant data themselves and associate the user by using ``Sponge.getServer().getGameProfileManager()``.
+But writing different code for offline and online users is not necessary:
+
+The Service Manager natively provides a service called ``UserStorageService`` which is capable of returning ``User`` instances for Players who are currently offline
+And since the ``Player`` interface extends ``User`` a ``User`` can often be used the same way as the ``Player`` instance.
+
+For example:
+
+* ``#hasPermission(String permission)`` is available from both instances.
+
+Code Example
+----------------------------------
+
+Getting a User instance is a few more lines more efford that getting a Player.
+Here's an example for a utility method that can be used to get a User:
+
+.. code-block:: java
```

Comment:
Usually for code examples we leave out all class declarations and method names. Just the relevant lines about using the user storage methods would be appropriate